File: backend/src/main/java/br/com/calciolari/datahub/imports/infrastructure/storage/RawStorageIntegrityException.java
package br.com.calciolari.datahub.imports.infrastructure.storage;

/**
 * Raised when an existing raw artifact disagrees on hash or size (no-clobber integrity).
 */
public class RawStorageIntegrityException extends RuntimeException {

	public RawStorageIntegrityException(String message) {
		super(message);
	}

	public RawStorageIntegrityException(String message, Throwable cause) {
		super(message, cause);
	}
}

File: backend/src/Calciolari.DataHub/Imports/Infrastructure/Storage/RawStorageIntegrityException.cs
namespace Calciolari.DataHub.Imports.Infrastructure.Storage;

/// <summary>
/// Raised when an existing raw artifact disagrees on hash or size (no-clobber integrity).
/// </summary>
public sealed class RawStorageIntegrityException : Exception
{
    public RawStorageIntegrityException(string message)
        : base(message)
    {
    }

    public RawStorageIntegrityException(string message, Exception inner)
        : base(message, inner)
    {
    }
}
